src/vm/frames.cpp

```diff
@@ -1001,6 +1001,31 @@ VOID GCFrame::Pop()
 #endif
 }
 
+#ifndef FEATURE_PAL
+// GCFrame destructor removes the GCFrame from the current thread's explicit frame list.
+// This prevents issues in functions that have HELPER_METHOD_FRAME_BEGIN / END around 
+// GCPROTECT_BEGIN / END and where the C++ compiler places some local variables over 
+// the stack location of the GCFrame local variable after the variable goes out of scope. 
+GCFrame::~GCFrame()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    if (m_Next != NULL)
```

Comment:
Would it make sense to get rid of the Pops in the GC_PROTECT macro and just depend on doing the Pop here?

Reply:
That makes sense